lib/src/install.rs

```diff
@@ -1298,13 +1304,20 @@ async fn install_with_sysroot(
     tracing::debug!("Installed bootloader");
 
     tracing::debug!("Perfoming post-deployment operations");
-    // Note that we *always* initialize this container storage, even
-    // if there are no bound images today.
-    let imgstore = sysroot.get_ensure_imgstore()?;
-    // Now copy each bound image from the host's container storage into the target.
-    for image in bound_images {
-        let image = image.image.as_str();
-        imgstore.pull_from_host_storage(image).await?;
+
+    if state.config_opts.pull {
+        crate::boundimage::pull_images(sysroot, bound_images)
+            .await
+            .context("pulling bound images")?;
+    } else {
+        // Note that we *always* initialize this container storage, even
+        // if there are no bound images today.
+        let imgstore = sysroot.get_ensure_imgstore()?;
```

Comment:
Don't we need this unconditionally i.e. outside the `if`?

Reply:
True, forgot it has side effects